src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -250,14 +250,14 @@ public static ReadableBuffer Create(byte[] data, int offset, int length)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
 
-            var buffer = new OwnedArray<byte>(data);
+            OwnedBuffer<byte> buffer = data;
```

Comment:
This is not about guidelines/conventions. var buffer = data would declare and assign T[] buffer local. We need buffer to be OwnedBuffer\<byte\> not byte[]